### PR TITLE
x11-misc/albert: version bump to 0.17.3

### PR DESCRIPTION
**Pull request by CaseOf**

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
I can't get this to compile...
```
/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -           lVBoxXPCOM: No such file or directory
collect2: error: ld returned 1 exit status
```

**Review by juippis — COMMENTED** [review]
I noticed this needs **dev-qt/qtxml** to build, too.

With `-virtualbox` I get:
```
In file included from /var/tmp/portage/x11-misc/albert-0.17.3/work/albert-0.17.3/plugins/python/        pybind11/include/pybind11/attr.h:13,
                 from /var/tmp/portage/x11-misc/albert-0.17.3/work/albert-0.17.3/plugins/python/        pybind11/include/pybind11/pybind11.h:45,
                 from /var/tmp/portage/x11-misc/albert-0.17.3/work/albert-0.17.3/plugins/python/        pybind11/include/pybind11/embed.h:12,
                 from /var/tmp/portage/x11-misc/albert-0.17.3/work/albert-0.17.3/plugins/python/src/    extension.cpp:3:
/var/tmp/portage/x11-misc/albert-0.17.3/work/albert-0.17.3/plugins/python/pybind11/include/pybind11/    cast.h: In function 'std::string pybind11::detail::error_string()':
/var/tmp/portage/x11-misc/albert-0.17.3/work/albert-0.17.3/plugins/python/pybind11/include/pybind11/    cast.h:446:36: error: invalid use of incomplete type 'PyFrameObject' {aka 'struct _frame'}
  446 |                 "  " + handle(frame->f_code->co_filename).cast<std::string>() +
      |                                    ^~
```

This seems to be related to `+python` flag now.

With `+virtualbox` after recompiling virtualbox with `+vboxwebsrv` it indeed works.

**Comment by CaseOf** [comment]
I thinx your python issue is tied to py3.11.
I have py3.10 and it is working.
I dont know how should I fix this issue

**Comment by CaseOf** [comment]
I could probably use packaged version to check.
How could I get dynamically site-packages path in an ebuild?

**Comment by CaseOf** [comment]
Hi, I tried with dev-python/pybind11 packaged version, can you review it?

File: x11-misc/albert/albert-0.17.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtsql:5[sqlite]
+	dev-qt/qtsvg:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtx11extras:5
+	python? ( ${PYTHON_DEPS} )
+	statistics? ( dev-qt/qtcharts:5 )
+	virtualbox? ( app-emulation/virtualbox[sdk] )
+	x11-libs/libX11
```
Comment (juippis) [inline x11-misc/albert/albert-0.17.3.ebuild:56]:
```suggestion
	dev-cpp/muParser
	dev-qt/qtconcurrent:5
	dev-qt/qtcore:5
	dev-qt/qtdbus:5
	dev-qt/qtdeclarative:5
	dev-qt/qtgraphicaleffects:5
	dev-qt/qtgui:5
	dev-qt/qtnetwork:5
	dev-qt/qtsql:5[sqlite]
	dev-qt/qtsvg:5
	dev-qt/qtwidgets:5
	dev-qt/qtx11extras:5
	x11-libs/libX11
	python? ( ${PYTHON_DEPS} )
	statistics? ( dev-qt/qtcharts:5 )
	virtualbox? ( app-emulation/virtualbox[sdk] )
```

Sort unconditional deps before conditional ones.

File: x11-misc/albert/albert-0.17.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	dev-qt/qtdeclarative:5
+	dev-qt/qtgraphicaleffects:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtsql:5[sqlite]
+	dev-qt/qtsvg:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtx11extras:5
+	python? ( ${PYTHON_DEPS} )
+	statistics? ( dev-qt/qtcharts:5 )
```
Comment (juippis) [inline x11-misc/albert/albert-0.17.3.ebuild:54]: you'll also need to depend on dev-qt/qtxml here.

File: x11-misc/albert/albert-0.17.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	virtualbox? ( app-emulation/virtualbox[sdk] )
```
Comment (juippis) [inline x11-misc/albert/albert-0.17.3.ebuild:55]: Still fails to compile without `+vboxwebsrv` here.

**Review by juippis — COMMENTED** [review]
It compiles now, but does it have to be this complicated? I figure if https://github.com/albertlauncher/plugins/pull/11/files is applied (may need a tiny modification for removing pybind::embed) you could get rid of the conditional in `src_prepare`? Or if there's a way to tell cmake where to find the python libs (I haven't cloned upstream code to grep).
 Could you try?

**Comment by CaseOf** [comment]
I did amend my commit to follow your advices.
Now, I’ll look at this upstream pull request to see if I can make it build.

**Comment by CaseOf** [comment]
I did try with this way for the patch, and it seems to work.
Can you confirm it?
```patch
From aa89c47db62163eddea623beb48f7ffd8d76062c Mon Sep 17 00:00:00 2001
From: Buchan Milne <bgmilne@gmail.com>
Date: Thu, 14 Jun 2018 08:10:31 +0200
Subject: [PATCH] Find and use python libraries and headers

---
 plugins/python/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plugins/python/CMakeLists.txt b/plugins/python/CMakeLists.txt
index 2dd085d..0dc9b91 100644
--- a/plugins/python/CMakeLists.txt
+++ b/plugins/python/CMakeLists.txt
@@ -5,7 +5,8 @@ PROJECT(python)
 FILE(GLOB_RECURSE SRC src/*)
 
 find_package(Qt5 5.5.0 REQUIRED COMPONENTS Widgets)
-add_subdirectory(pybind11)
+find_package(PythonLibs)
+include_directories(${PYTHON_INCLUDE_DIRS})
 
 add_library(${PROJECT_NAME} SHARED ${SRC} ${PROJECT_NAME}.qrc metadata.json)
 
@@ -14,8 +15,8 @@ target_include_directories(${PROJECT_NAME} PRIVATE src/)
 target_link_libraries(${PROJECT_NAME}
     PUBLIC
         Qt5::Widgets
+        ${PYTHON_LIBRARIES}
     PRIVATE
-        pybind11::embed
         albert::lib
         xdg
 )

```
with the ebuild below:
```ebuild
# Copyright 1999-2022 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=8

PYTHON_COMPAT=( python3_{8..11} )

inherit cmake python-single-r1 xdg-utils

DESCRIPTION="A fast and flexible keyboard launcher"
HOMEPAGE="https://albertlauncher.github.io/"

PLUGINS_HASH="ee55048e138028b4889d71e0574e85b2c4d69541"
PYBIND11_VERSION="2.6.1"
PYTHON_EXTENSIONS_COMMIT="7f571aceaf8e60eee8bb21e1ec4efa0e95523d13"
JETBRAINS_PYTHON_EXTENSION_COMMIT="b7157473cc923fe4f15023c85a032eeab3627652"
XKCD_PYTHON_EXTENSION_COMMIT="bf88a964473d65b39c9e09eb48dabb847206f06f"

SRC_URI="
	https://github.com/albertlauncher/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
	https://github.com/albertlauncher/plugins/archive/${PLUGINS_HASH}.tar.gz -> ${P}-plugins.tar.gz
	python? (
		https://github.com/pybind/pybind11/archive/v${PYBIND11_VERSION}.tar.gz -> ${P}-pybind11-${PYBIND11_VERSION}.tar.gz
	)
	python-extensions? (
		https://github.com/albertlauncher/python/archive/${PYTHON_EXTENSIONS_COMMIT}.tar.gz -> ${P}-python-extensions.tar.gz
		https://github.com/mqus/jetbrains-albert-plugin/archive/${JETBRAINS_PYTHON_EXTENSION_COMMIT}.tar.gz -> ${P}-jetbrains-python-extension.tar.gz
		https://github.com/bergercookie/xkcd-albert-plugin/archive/${XKCD_PYTHON_EXTENSION_COMMIT}.tar.gz -> ${P}-xkcd-python-extension.tar.gz
	)
"

LICENSE="all-rights-reserved"	# unclear licensing #766129
SLOT="0"
KEYWORDS="~amd64"
IUSE="debug +python +python-extensions +statistics virtualbox"
RESTRICT="mirror bindist"

REQUIRED_USE="
	python-extensions? ( python )
	python? ( ${PYTHON_REQUIRED_USE} )
"

RDEPEND="
	dev-cpp/muParser
	dev-qt/qtconcurrent:5
	dev-qt/qtcore:5
	dev-qt/qtdbus:5
	dev-qt/qtdeclarative:5
	dev-qt/qtgraphicaleffects:5
	dev-qt/qtgui:5
	dev-qt/qtnetwork:5
	dev-qt/qtsql:5[sqlite]
	dev-qt/qtsvg:5
	dev-qt/qtwidgets:5
	dev-qt/qtx11extras:5
	x11-libs/libX11
	python? ( ${PYTHON_DEPS} )
	statistics? (
		dev-qt/qtcharts:5
		dev-qt/qtxml:5
	)
	virtualbox? ( app-emulation/virtualbox[sdk,vboxwebsrv] )
"
DEPEND="${RDEPEND}
	python? ( $(python_gen_cond_dep 'dev-python/pybind11[${PYTHON_USEDEP}]') )"

PATCHES=("${FILESDIR}/${P}-find-and-use-python-libraries-and-headers.patch")

src_prepare() {
	mv "${WORKDIR}"/plugins-${PLUGINS_HASH}/* "${S}"/plugins || die
	if use python-extensions; then
		mv "${WORKDIR}"/python-${PYTHON_EXTENSIONS_COMMIT}/* "${S}"/plugins/python/share/modules || die
		mv "${WORKDIR}"/jetbrains-albert-plugin-${JETBRAINS_PYTHON_EXTENSION_COMMIT}/* "${S}"/plugins/python/share/modules/jetbrains_projects || die
		mv "${WORKDIR}"/xkcd-albert-plugin-${XKCD_PYTHON_EXTENSION_COMMIT}/* "${S}"/plugins/python/share/modules/xkcd || die
	fi

	cmake_src_prepare
}

src_configure() {
	local mycmakeargs=(
		-DBUILD_DEBUG=$(usex debug)
		-DBUILD_PYTHON=$(usex python)
		-DBUILD_VIRTUALBOX=$(usex virtualbox)
		-DBUILD_WITH_QTCHARTS=$(usex statistics)
	)

	cmake_src_configure
}

pkg_postinst() {
	xdg_desktop_database_update
	xdg_icon_cache_update
}

pkg_postrm() {
	xdg_desktop_database_update
}

```

**Comment by CaseOf** [comment]
pushed my changes on the pull request

**Comment by CaseOf** [comment]
removed remaining pybind11 files in SRC_URI

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**